kustomize/commands/edit/add/flagsandargs_test.go

```diff
@@ -149,3 +149,38 @@ func TestExpandFileSourceWithKeyAndError(t *testing.T) {
 		t.Fatalf("FileSources should not be correctly expanded: %v", fa.FileSources)
 	}
 }
+
+func TestExpandFileSourceWithHiddenFiles(t *testing.T) {
+	fSys := filesys.MakeEmptyDirInMemory()
+	_, err := fSys.Create("dir/file-1.xtn")
+	require.NoError(t, err)
```

Comment:
Suggestion: instead of repeating the create calls and error checking, use a loop over a list of files you want to create:

```go
	files := []string{
		"dir/file-1.xtn",
		"dir/file-2",
		"dir/.file-3",
		"dir/dir-2/.file-4",
		"dir/dir-2/file-5.xtn",
		"dir/dir-2/.file-6.xtn",
		"root-1.xtn",
		".root-2.xtn",
	}
	for _, file := range files {
		_, err := fSys.Create(file)
		require.NoError(t, err)
	}
```